### PR TITLE
BreadcrumbsHelper documentation

### PR DESCRIPTION
**Pull request by HavokInspiration**
This is a first draft for the BreadcrumbsHelper documentation.
There is most certainly room for improvements (and probably build fixes :smile:)
Once we have something approved in English, I'll take care of the French translation.

Refs https://github.com/cakephp/cakephp/pull/9562

### CODE REVIEW

File: en/views/helpers/breadcrumbs.rst [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    );
+
+    // Prepended middleware will be put at the top of the list
+    $this->Breadcrumbs->prepend(
+        'Products',
+        ['controller' => 'products', 'action' => 'index']
+    );
+
+    // Insert in a specific slot. If the slot is out of
+    // bounds, it will be added to the end.
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: Why isn't an exception raised as with ``insertBefore()``?

File: en/views/helpers/breadcrumbs.rst [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    $this->Breadcrumbs->insertAt(
+        2,
+        'Products',
+        ['controller' => 'products', 'action' => 'index']
+    );
+
+    // Insert before another crumb, based on the title.
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: What if the title in question exists more than once?

File: en/views/helpers/breadcrumbs.rst [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+precisely where you want to add a crumb.
+
+Rendering the breadcrumbs trail
+===============================
+
+Basic rendering
+---------------
+
+After adding crumbs to the trail, you can easily render it using the
+``render()`` methods.
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: "methods", plural?

File: en/views/helpers/breadcrumbs.rst [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+This method accepts two arguments, both of them arrays:
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: "This method accepts two array arguments."

File: en/views/helpers/breadcrumbs.rst [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        ['separator' => '<i class="fa fa-angle-right"></i>']
+    );
+
+Customizing the output
+----------------------
+
+Customizing the templates
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The BreadcrumbsHelper internally uses the ``StringTemplateTrait`` which gives
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: Comma: ``StringTemplateTrait``, which ...

File: en/views/helpers/breadcrumbs.rst [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+You can easily customize them using the ``templates()`` method from the
+``StringTemplateTrait``::
+
+    $this->Breadcrumbs->templates([
+        'wrapper' => '<nav class="breadcrumbs"><ul{{attrs}}>{{content}}</ul></nav>',
+    ]);
+
+Since your templates will be rendered supporting the ``templateVars``, you can
+add your own templates var in the various templates::
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: "templates var" -> "template variables"?

File: en/views/helpers/breadcrumbs.rst [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+        [
+            'templateVars' => [
+                'icon' => '<i class="fa fa-money"></i>'
+            ]
+        ]
+    );
+
+.. _defining_attributes_item:
+
+Defining attributes for the item
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: Defining Attributes for the Item

File: en/views/helpers/breadcrumbs.rst [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+    echo $this->BreadcrumbsHelper->render(
+        ['class' => 'breadcrumbs-trail'],
+        ['separator' => '<i class="fa fa-angle-right"></i>']
+    );
+
+Customizing the output
+----------------------
+
+Customizing the templates
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: Customizing the Templates

File: en/views/helpers/breadcrumbs.rst [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+  (empty), it will not render a separator.
+
+Here is an example of how to render a trail::
+
+    echo $this->BreadcrumbsHelper->render(
+        ['class' => 'breadcrumbs-trail'],
+        ['separator' => '<i class="fa fa-angle-right"></i>']
+    );
+
+Customizing the output
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: Customizing the Output

File: en/views/helpers/breadcrumbs.rst [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+Using these methods give you the ability to go around CakePHP's way of
+rendering the view. Since templates and layouts are rendered from the inside
+out (meaning, included elements are rendered first), this allows you to define
+precisely where you want to add a crumb.
+
+Rendering the breadcrumbs trail
+===============================
+
+Basic rendering
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: Basic Rendering

File: en/views/helpers/breadcrumbs.rst [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+        'Products',
+        ['controller' => 'products', 'action' => 'index']
+    );
+
+Using these methods give you the ability to go around CakePHP's way of
+rendering the view. Since templates and layouts are rendered from the inside
+out (meaning, included elements are rendered first), this allows you to define
+precisely where you want to add a crumb.
+
+Rendering the breadcrumbs trail
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: Rendering the Breadcrumbs Trail

File: en/views/helpers/breadcrumbs.rst [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+.. php:namespace:: Cake\View\Helper
+
+.. php:class:: BreadcrumbsHelper(View $view, array $config = [])
+
+.. versionadded:: 3.3.6
+
+BreadcrumbsHelper provides a way to easily deal with the creation and rendering
+of a breadcrumbs trail for your app.
+
+Creating a breadcrumbs trail
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: Creating a Breadcrumbs Trail

File: en/views/helpers/breadcrumbs.rst [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+            ]
+        ]
+    );
+
+.. _defining_attributes_item:
+
+Defining attributes for the item
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to apply specific HTML attributes to both the item and its sub-item
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: Comma: "both, the"

File: en/views/helpers/breadcrumbs.rst [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+, you can leverage the ``innerAttrs`` key the ``$options`` argument gives you.
```
Comment (ravage84) [inline en/views/helpers/breadcrumbs.rst]: "key, which the ``$options`` argument provides you."?

File: en/views/helpers/breadcrumbs.rst (hunk 2 above)
Comment (HavokInspiration) [inline en/views/helpers/breadcrumbs.rst]: First matched is used. That would be super weird though to have that.

File: en/views/helpers/breadcrumbs.rst (hunk 13 above)
Comment (HavokInspiration) [inline en/views/helpers/breadcrumbs.rst]: I'm not sure about this one. This breaks the flow.
If I start my sentence with that, there would be no comma. "Both the item and its sub-items [...]".

**Comment by HavokInspiration** [comment]
@ravage84 Thanks for the review. Looking at different pages, titles seems to be capitalized, so I'll just follow what it is in place.

File: en/views/helpers/breadcrumbs.rst (hunk 1 above)
Comment (HavokInspiration) [inline en/views/helpers/breadcrumbs.rst]: This should be discussed in https://github.com/cakephp/cakephp/pull/9562

**Comment by HavokInspiration** [comment]
I pushed some commits :
- changes after review
- added the files to the TOC
- added the French translation
- some minor fixes in the english doc

**Comment by markstory** [comment]
Pushed up my changes as a commit.

**Comment by HavokInspiration** [comment]
@markstory Thanks for the grammar.

**Comment by HavokInspiration** [comment]
Updated the doc with the new changes from the PR on the framework repo.

**Pull request merged**